### PR TITLE
rewrite index create and delete feature, now scans from bottom to top

### PR DESCRIPTION
**Pull request by shokakucarrier**

### CODE REVIEW

File: mrrc/pkgs/indexing.py [hunk 1]
```diff
@@ -42,212 +40,217 @@ def get_index_template() -> str:
...
         if items == set():
-            delete_index.append(local_index_file)
-            if path != '':
-                deleted_files.add(path)
-            if local_index_file in update_index:
-                update_index.remove(local_index_file)
-        # if there is still items left, save it on local file, and mark it has already been updated
+            with open(local_index_file, 'a+', encoding='utf-8') as f:
+                f.write(item + '/\n')
+            if dir_index != '.index':
```
Comment (ligangty) [inline mrrc/pkgs/indexing.py:153]: Should this condition happen before line#149? If it is not an .index, why should it be loaded from s3 again?

File: mrrc/pkgs/indexing.py [hunk 2]
```diff
@@ -42,212 +40,217 @@ def get_index_template() -> str:
...
+            virtual_folder_delete(os.path.dirname(path), base_dir, s3_client, bucket,
+                                  updated_indexes, delete_indexes)
+    else:
+        items = load_s3_index(s3_client, bucket, dir_index)
+        with open(local_index_file, 'w+', encoding='utf-8') as f:
+            letf_items = items.difference({item})
+            if letf_items == set():
+                updated_indexes.remove(dir_index)
+                delete_indexes.add(dir_index)
+                if dir_index != '.index':
```
Comment (ligangty) [inline mrrc/pkgs/indexing.py:202]: same as line#153.

**Review by ligangty — COMMENTED** [review]
Some comments.

File: test.sh [hunk 3]
```diff
@@ -2,7 +2,7 @@
 set -eux
 
 # Prepare env vars
-ENGINE=${ENGINE:="podman"}
+ENGINE=${ENGINE:="docker"}
```
[the remark this answers is not in the document]
Comment (shokakucarrier) [reply test.sh:5]: SORRY!!!

**Review by ligangty — APPROVED** [review]
Looks good. But still one bug of the root index.html deletion problem. Will merge this and see the fix in new PR.

**Pull request merged**